### PR TITLE
Adding Scaffolding for the Editor Backend

### PR DESCRIPTION
**Pull request by Varun-Sethu**
This entire PR is just introducing some scaffolding for the editor backend, I feel like its prone to deadlocks just like the old one

### CODE REVIEW

File: Backend/editor/client.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package editor
+
+import (
+	"github.com/gorilla/websocket"
+)
+
+// client is the embodyment of all data relating to a client connection
```
Comment (angary) [inline Backend/editor/client.go:7]:
```suggestion
// client is the embodiment of all data relating to a client connection
```

File: Backend/editor/main.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import (
+	"log"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
+)
+
+// This file just defines some of the endpoints for the editor
+// and ties togher its various disparate components
```
Comment (angary) [inline Backend/editor/main.go:12]:
```suggestion
// and ties together its various disparate components
```

File: Backend/editor/server.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	clients   map[int]*clientState
+}
+
+type clientState struct {
+	*client
+	canSendOps bool
+}
+
+func newServer() *server {
+	// ideally state shouldnt be a string due to its immutability
```
Comment (angary) [inline Backend/editor/server.go:24]:
```suggestion
	// ideally state shouldn't be a string due to its immutability
```

**Review by angary — CHANGES_REQUESTED** [review]

**Comment by angary** [comment]
lgtm

**Comment by angary** [comment]
LGTM!!!

**Review by brianzhang2101 — APPROVED** [review]
lgtm (let's get this money)

**Pull request merged**